relay-server/src/utils/metrics_rate_limits.rs

```diff
@@ -130,112 +131,212 @@ pub enum ExtractionMode {
     Duration,
 }
 
-/// Return value of [`count_metric_bucket`], containing the extracted
-/// count of transactions and wether they have associated profiles.
-#[derive(Debug, Clone, Copy)]
-struct TransactionCount {
-    /// Number of transactions.
-    pub count: usize,
-    /// Whether the transactions have associated profiles.
-    pub has_profile: bool,
+/// The return value of [`count_metric_bucket`].
+///
+/// Contains the count of total transactions or spans that went into this bucket.
+#[derive(Debug, Default, Clone)]
+pub enum BucketSummary {
+    Transactions {
+        count: usize,
+        has_profile: bool,
+    },
+    Spans(usize),
+    #[default]
+    None,
+}
+
+impl BucketSummary {
+    fn to_counts(&self) -> TotalEntityCounts {
+        match *self {
+            BucketSummary::Transactions { count, has_profile } => TotalEntityCounts {
+                transactions: Some(count),
+                spans: None,
+                profiles: if has_profile { count } else { 0 },
+            },
+            BucketSummary::Spans(count) => TotalEntityCounts {
+                transactions: None,
+                spans: Some(count),
+                profiles: 0,
+            },
+            BucketSummary::None => TotalEntityCounts::default(),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct SummarizedBucket {
+    bucket: Bucket,
+    summary: BucketSummary,
+}
+
+#[derive(Debug, Default, Clone)]
+struct TotalEntityCounts {
+    transactions: Option<usize>,
```

Comment:
Why is `profiles` `usize` but all the others `Option<usize>`, also technically `transactions: Some(0)` is allowed here, shouldn't this just be an `usize`?

I think that simplifies a bunch of code.

Reply:
They mean different things:

1. `None` means there are no transaction-related metrics at all, so we don't even have to check any rate limits.
2. `Some(0)` means there are transaction-related metrics in the batch that do not increase the redis counter, but still need to be removed when a quota is active.

I will add more docs to clarify.

Reply:
But do we actually need this, I only see you act on the Some/None part for rate limits?

Reply:
-> We need the distinction 👍